src/backend/postmaster/pgstat.c

```diff
@@ -1792,65 +1760,101 @@ get_tabstat_entry(Oid rel_id, bool isshared)
 	TabStatusArray *tsa;
 	bool found;
 
-	make_sure_stat_tab_initialized();
+	/*
+	 * Create hash table if we don't have it already.
+	 */
+	if (pgStatTabHash == NULL)
+	{
+		HASHCTL			ctl;
+
+		memset(&ctl, 0, sizeof(ctl));
+		ctl.keysize = sizeof(Oid);
+		ctl.entrysize = sizeof(TabStatHashEntry);
```

Comment:
You should backport working commit!
```suggestion
		ctl.entrysize = sizeof(TabStatHashEntry);
		ctl.hash = oid_hash;
```